class.jetpack.php

```diff
@@ -3188,14 +3189,90 @@ public static function get_activation_source( $referer_url ) {
 	/**
 	 * Runs before bumping version numbers up to a new version
 	 *
-	 * @param  string $version    Version:timestamp
-	 * @param  string $old_version Old Version:timestamp or false if not set yet.
-	 * @return null              [description]
+	 * @param string $version    Version:timestamp.
+	 * @param string $old_version Old Version:timestamp or false if not set yet.
 	 */
 	public static function do_version_bump( $version, $old_version ) {
-		if ( ! $old_version ) { // For new sites
-			// There used to be stuff here, but this seems like it might  be useful to someone in the future...
+		if ( $old_version ) { // For existing Jetpack installations.
+			self::send_update_modal_data();
+		}
+	}
+
+	/**
+	 * Prepares the release post content and image data and saves it in the
+	 * state array. This data is used to create the update modal.
+	 */
+	public static function send_update_modal_data() {
+		$post_data = self::get_release_post_data();
+
+		if ( ! isset( $post_data ) || ! isset( $post_data['posts'] ) || ! isset( $post_data['posts'][0] ) ) {
+			return;
+		}
+
+		$post_content = isset( $post_data['posts'][0]['content'] ) ? $post_data['posts'][0]['content'] : null;
+		if ( empty( $post_content ) ) {
+			return;
+		}
+
+		$post_image_src = isset( $post_data['posts'][0]['featured_image'] ) ? $post_data['posts'][0]['featured_image'] : null;
+		$post_link      = isset( $post_data['posts'][0]['URL'] ) ? $post_data['posts'][0]['URL'] : null;
+
+		$post_content = str_replace( '<a ', '<a target="_blank" rel="noopener noreferrer" ', $post_content );
+		$content      = wp_kses_post( $post_content );
+		$image_src    = esc_url( $post_image_src );
+		$link         = esc_url( $post_link );
+
+		$post_array = array(
+			'release_post_content' => $content,
+			'release_post_image'   => $image_src,
+			'release_post_link'    => $link,
+		);
+
+		self::state( 'message_content', $post_array );
+	}
+
+	/**
+	 * Obtains the most recent post in the 'releases' category in the Jetpack.com
+	 * blog. Returns an array with the release post's data at index ['posts'][0].
+	 *
+	 * The response parameters for the post array can be found here:
+	 * https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/%24post_ID/#apidoc-response
+	 *
+	 * @return array|null Returns an associative array containing the release post data.
+	 *                    Returns null if the release post data is not available.
+	 */
+	private static function get_release_post_data() {
+		if ( Constants::is_defined( 'TESTING_IN_JETPACK' ) && Constants::get_constant( 'TESTING_IN_JETPACK' ) ) {
+			return null;
 		}
+		/* Need to add tags to the release posts to use this.
+		$version = JETPACK__VERSION;
+		$release_post_src = add_query_arg(
+			array(
+				'order_by' => 'date',
+				'tag'      => $version,
+				'number'   => '1',
+			),
+			'https://public-api.wordpress.com/rest/v1/sites/jetpack.com/posts'
+		);
+		*/
+
```

Comment:
We can switch to using the tag in the release post query. We just need to decide on the exact tag that we'll use with the release posts.

Reply:
Would the actual version number be okay, e.g. `7.9`?

Reply:
Yes, I think that will be okay!

Reply:
This is done - see commit 4d71be6.